File: helidon-server/src/main/java/com/programacion/web/db/Album.java
package com.programacion.web.db;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Album {

    private Integer id;
    private String title;
    private User user;
    private List<Photo> photos;

}

File: FrontendReact/src/models/Album.ts
import type { User } from "./User";

export interface Album {
    id: number;
    title: string;
    user?: User;
    photos?: Photo[];
}

export interface Photo {
    id: number;
    title: string;
    url: string;
    thumbnailUrl: string;
    album?: Album;
}
